clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -99,19 +100,30 @@ void CIRGenModule::emitGlobal(clang::GlobalDecl gd) {
 void CIRGenModule::emitGlobalFunctionDefinition(clang::GlobalDecl gd,
```

Comment:
Does it make sense to pass `gd` as `clang::GlobalDecl` when it is always `clang::FunctionDecl`. It would eliminate multiple casts using `clang::FunctionDecl` directly as function codegen parameters.

Reply:
The equivalent function in Clang CodeGen has a `GlobalDecl` parameter, and there are some calls to that function where a `FunctionDecl` is not readily available.  So I'm not anxious to make this change right now.